adagrams/game.py

```diff
@@ -1,11 +1,152 @@
+  
+        for letter in hand_dict.keys():
+            if hand_dict[letter] <= LETTER_POOL[letter]:
+                valid_letter_count += 1
+
+        if valid_letter_count == 10:
+            valid_hand = True
+        else:
+            hand = random.choices(letters, weights, k = num_letters) 
+
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = list(word.upper())
+    letter_bank = list("".join(letter_bank).upper())
+    matched_letters = []
+  
+    if len(word) <= len(letter_bank):
+        for letter in letter_bank:
+            if word.count(letter) > letter_bank.count(letter):
+                return False
+
+        for letter in word:
+           if letter in letter_bank:
+               matched_letters.append(letter)
+           else:
+               return False
 
+        if len(word) == len(matched_letters):
+            for letter in word:
+                if word.count(letter) <= matched_letters.count(letter):
+                    continue
+                else:
+                    return False
+            return True
+        else:
+            return False  
+    else:
+        return False
+    
 def score_word(word):
-    pass
+    score_value_dict = {
+                    'A': 1, 
+                    'B': 3, 
+                    'C': 3, 
+                    'D': 2, 
+                    'E': 1, 
+                    'F': 4, 
+                    'G': 2, 
+                    'H': 4, 
+                    'I': 1, 
+                    'J': 8, 
+                    'K': 5, 
+                    'L': 1, 
+                    'M': 3, 
+                    'N': 1, 
+                    'O': 1, 
+                    'P': 3, 
+                    'Q': 10, 
+                    'R': 1, 
+                    'S': 1, 
+                    'T': 1, 
+                    'U': 1, 
+                    'V': 4, 
+                    'W': 4, 
+                    'X': 8, 
+                    'Y': 4, 
+                    'Z': 10
+}
+    word = list(word.upper())
+    score = 0
+
+    if len(word) >= 7:
+        score += 8
+
+    for letter in word:
+        if letter in score_value_dict:
+            score += score_value_dict[letter]
+    
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    highest_word_score = 0
+    winning_word = ""
+    words_len_ten = []
+
+    for word in word_list:
+        score = score_word(word)
+        if score == highest_word_score:
+            if len(word) == 10:
+                words_len_ten.append(word)
+            elif len(word) < len(winning_word) and not len(winning_word) == 10:
+                winning_word = word
+        elif score > highest_word_score:
+            highest_word_score = score
+            winning_word = word
+            if len(word) == 10:
+                words_len_ten.append(word)
+    
+    if words_len_ten:
+        return (words_len_ten[0], highest_word_score)
```

Comment:
If we have several words tied for highest, then we should return the first of those high scores with 10 letters, if there is one. If we have input like `["AAAAAAAAAD", "JQ", "KFHK"]` (In order, their scores are 11, 18, and 18), then we should still return `JQ` as the shortest of the highest scored words. The data in our tests didn't catch this scenario, but how could we update the function to account for it?